### DISCUSSION TITLE
update nested object value entire

### DISCUSSION THREAD

**Discussion by yjkwon07** [react-hook-form/react-hook-form#9162] [Q&A] [answered]
When I use setValue for object reset, not trigger register value. 😕

[codesandbox](https://codesandbox.io/s/react-hook-form-object-change-lcl7b2?file=/src/index.js)

```jsx
const defaultValues = {
  name: { site: "vanilla", code: "ss" }
};

...

<button
  className="button buttonBlack"
  type="button"
  onClick={() => {
    setValue("name", undefined);
  }}
>

...

<input {...register("name.site")} />
```

I found solution1 just update key name value. It's fine.

```jsx

<button
  className="button buttonBlack"
  type="button"
  onClick={() => {
    setValue("name.site", undefined);
  }}
>

<input {...register("name.site")} />
```

- but, I want entire object reset, trigger too.. 😢

**Comment by bluebill1049** [comment, marked as answer]
answerd here: https://github.com/react-hook-form/react-hook-form/discussions/9164
